app/hoc/withAuth/index.jsx

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect, withRouter } from 'react-router-dom';
+import { paths } from 'helpers/routes';
+
+export default function withAuth(WrappedComponent) {
+  class EnhancedComponent extends Component {
+    static propTypes = {
+      loggedIn: PropTypes.bool.isRequired
+    };
+
+    render() {
+      if (this.props.loggedIn) {
+        return (
+          <WrappedComponent { ...this.props } />
+        );
+      }
+
+      return <Redirect to={ paths.home() } />;
+    }
+  }
+
+  return withRouter(connect(state => state.session)(EnhancedComponent));
```

Comment:
I'm not sure, but seems like we don't need `withRouter` hoc here. More correctly as I think use it separately where it really is needed, like a `withAuth(withRouter(Component))`